### PR TITLE
IGAPP-676: Move rendering of layout to routes

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web-ts/src/routes/RootSwitcher.tsx [hunk 1]
```diff
@@ -30,23 +43,26 @@ import LoadingSpinner from '../components/LoadingSpinner'
 import { useTranslation } from 'react-i18next'
 
 export const cityContentPattern = `/:cityCode/:languageCode`
+type cityContentPatternType = typeof cityContentPattern
 
 export const RoutePatterns = {
-  [LANDING_ROUTE]: `/${LANDING_ROUTE}/:languageCode`,
-  [MAIN_DISCLAIMER_ROUTE]: `/${MAIN_DISCLAIMER_ROUTE}`,
-  [NOT_FOUND_ROUTE]: `/${NOT_FOUND_ROUTE}`,
+  [LANDING_ROUTE]: `/${LANDING_ROUTE}/:languageCode` as `/${LandingRouteType}/:languageCode`,
```
Comment (klinzo) [inline web-ts/src/routes/RootSwitcher.tsx:49]: The type of template strings are always just `string`. By adding types here we have type security in the components (i.e. which parameters are available for the route, e.g. `cityCode` or `eventId`).

Comment (Taggotty) [reply web-ts/src/routes/RootSwitcher.tsx:49]: you could do the same as @SteffiStuffel did in her previous PR and declare RoutePatterns `as const` this way typescript is also able to correctly infer the corresponding types and all values would become readonly.

**Review by Taggotty — APPROVED** [review]

File: web-ts/src/routes/RootSwitcher.tsx (hunk 1 above)
Comment (klinzo) [reply web-ts/src/routes/RootSwitcher.tsx:49]: :pray: thanks! Works like a charm! Applied in #312

Comment (maxammann) [reply web-ts/src/routes/RootSwitcher.tsx:49]: Very cool! :)

File: web-ts/src/routes/sprungbrett/SprungbrettOfferPage.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { ReactElement } from 'react'
```
Comment (maxammann) [inline web-ts/src/routes/sprungbrett/SprungbrettOfferPage.tsx:1]: I'm not sure whether we discussed this somewhere already, but would it be possible to remove this one layer of directories here?

To have just a directory with an index.ts: `src/routes/SprungbrettOfferPage/index.ts`
or just have a single js file per route? `src/routes/SprungbrettOfferPage.ts?

**Review by maxammann — APPROVED** [review]

File: web-ts/src/routes/sprungbrett/SprungbrettOfferPage.tsx (hunk 2 above)
Comment (klinzo) [reply web-ts/src/routes/sprungbrett/SprungbrettOfferPage.tsx:1]: Done in #311, you just reviewed it :D

Comment (maxammann) [reply web-ts/src/routes/sprungbrett/SprungbrettOfferPage.tsx:1]: oh oke :D quite a lot of refactoring :P

**Pull request merged**